pkg/cluster/internal/providers/docker/util.go

```diff
@@ -48,15 +48,29 @@ func usernsRemap() bool {
 }
 
 // mountDevMapper checks if the Docker storage driver is Btrfs or ZFS
+// or if the backing filesystem is Btrfs
 func mountDevMapper() bool {
 	storage := ""
+	// check the docker storage driver
 	cmd := exec.Command("docker", "info", "-f", "{{.Driver}}")
 	lines, err := exec.OutputLines(cmd)
-	if err != nil {
+	if err != nil || len(lines) != 1 {
 		return false
 	}
-	if len(lines) > 0 {
-		storage = strings.ToLower(strings.TrimSpace(lines[0]))
+
+	storage = strings.ToLower(strings.TrimSpace(lines[0]))
+	if storage == "btrfs" || storage == "zfs" || storage == "devicemapper" {
+		return true
+	}
+
+	// check the backing file system
+	// docker info -f '{{json .DriverStatus  }}'
+	// [["Backing Filesystem","extfs"],["Supports d_type","true"],["Native Overlay Diff","true"]]
+	cmd = exec.Command("docker", "info", "-f", "{{ index (index .DriverStatus 0) 1 }}")
```

Comment:
do we have reason to believe this is always index 0 of DriverStatus?

Reply:
I tested it in 3 different deployments, maybe @AkihiroSuda can help us here ^^^

Reply:
This seems unnecessarily fragile we can decode the structure and inspect the tuple with "Backing Filesystem"